### PR TITLE
kv, puller (ticdc):  eliminating unnecessary goroutines, streamlining the kvClient's operation, and reducing code complexity.

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: ref #10879

### What is changed and how it works?

1. Renaming and Adjusting Data Structures: The singleRegionInfo structure has been renamed to regionInfo, and some of its fields have been adjusted. For example, requestedTable has been replaced with subscribedTable, which helps to more accurately reflect the purpose and content of the structure.

2. Function Signature and Logic Updates: Multiple functions have had their signatures modified to use the new regionInfo structure instead of the old singleRegionInfo. The internal logic of the functions has also been updated to accommodate the new data structures and naming.

3. Code Refactoring and Optimization: Redundant goroutines have been removed, and the code logic of kvClient has been simplified to make the code more clear and efficient.

#### Tests 

 - No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/kv/region_state.go [hunk 1]
```diff
@@ -31,33 +31,56 @@ const (
 	stateRemoved uint32 = 2
 )
 
-type singleRegionInfo struct {
-	verID  tikv.RegionVerID
+type regionInfo struct {
+	verID tikv.RegionVerID
+	// The span of the region.
+	// Note: The span is not always the real span of the region, it is the span of the
```
Comment (3AceShowHand) [inline cdc/kv/region_state.go:37]: this comment is not easy to understand.

File: cdc/kv/shared_client.go [hunk 2]
```diff
@@ -476,57 +491,50 @@ func (s *SharedClient) requestStore(
...
-func (s *SharedClient) broadcastRequest(r *requestedStore, sri singleRegionInfo) {
+func (s *SharedClient) broadcastRequest(r *requestedStore, region regionInfo) {
 	for _, stream := range r.streams {
-		stream.requests.In() <- sri
+		stream.requests.In() <- region
 	}
 }
 
-func (s *SharedClient) handleRequestRanges(ctx context.Context) error {
+func (s *SharedClient) handleRangeTask(ctx context.Context) error {
```
Comment (hicqu) [inline cdc/kv/shared_client.go:514]: Maybe `handleRangeTasks` is better.

File: cdc/puller/multiplexing_puller.go [hunk 3]
```diff
@@ -254,30 +322,38 @@ func (p *MultiplexingPuller) run(ctx context.Context, includeClient bool) error
 			zap.String("changefeed", p.changefeed.ID))
 	}()
 
-	g, ctx := errgroup.WithContext(ctx)
```
Comment (3AceShowHand) [inline cdc/puller/multiplexing_puller.go:257]: I prefer `g`.

File: cdc/kv/shared_client.go [hunk 4]
```diff
@@ -540,119 +548,129 @@ func (s *SharedClient) divideAndScheduleRegions(
...
 ) {
 	select {
-	case s.requestRangeCh.In() <- rangeTask{span: span, requestedTable: requestedTable}:
+	case s.rangeTaskCh.In() <- rangeTask{span: span, subscribedTable: subscribedTable}:
 	case <-ctx.Done():
 	}
 }
 
-func (s *SharedClient) handleErrors(ctx context.Context) error {
+func (s *SharedClient) handleError(ctx context.Context) error {
```
Comment (hicqu) [inline cdc/kv/shared_client.go:656]: Maybe `handleErrors` is better. It's a goroutine, not a single call for one region.

File: cdc/kv/shared_stream.go [hunk 5]
```diff
@@ -75,10 +75,10 @@ func newStream(ctx context.Context, c *SharedClient, g *errgroup.Group, r *reque
 			select {
 			case <-ctx.Done():
 				return ctx.Err()
-			case sri := <-stream.requests.Out():
-				if sri.lockedRange != nil {
-					stream.preFetchForConnecting = new(singleRegionInfo)
-					*stream.preFetchForConnecting = sri
+			case region := <-stream.requests.Out():
+				if region.lockedRange != nil {
```
Comment (hicqu) [inline cdc/kv/shared_stream.go:79]: Should use `isStopped`. Please also grep and replace other places.

**Comment by hicqu** [comment]
Rest LGTM.

File: cdc/kv/region_state.go (hunk 1 above)
Comment (CharlesCheung96) [reply cdc/kv/region_state.go:37]: Please add comments to explain that this situation is possible only when a single region contains data from multiple tables.

File: cdc/puller/multiplexing_puller.go (hunk 3 above)
Comment (asddongmen) [reply cdc/puller/multiplexing_puller.go:257]: The term "eg" is used elsewhere in the codebase, so I chose to maintain a consistent style here.

Comment (3AceShowHand) [reply cdc/puller/multiplexing_puller.go:257]: `g` is also used elsewhere, it's not a big matter, just ignore it. looks good to me

**Review by CharlesCheung96 — APPROVED** [review]

**Review by hicqu — APPROVED** [review]

**Pull request merged**